### DISCUSSION TITLE
react-table add column on button click

### DISCUSSION THREAD

**Discussion by HarshilPatel007** [TanStack/table#3985] [Q&A]
hello all,
I've created a table and added a button "Add Row". when user clicks on this button it'll add a new editable row to table. (this is done)
So, my question is, how can I make something like this but for columns?
like, "Add Column". when user clicks on it, it'll create a new column and adds empty fields to all available rows.
example,
- default table

| id | name |
|---|--------|
| 1 | john   |
| 2 | nick   |

- when user clicks on "Add column" button and gives column name "age"

| id | name | age |
|---|--------|------|
| 1 | john   ||
| 2 | nick   ||

**Comment by evans-tim** [comment]
If your columns are defined as component state, you can pass the columns as props to your table (directly to your table without a useMemo.) Every time the columns are updated, your table should rerender.
